compiler/src/iree/compiler/Dialect/LinalgExt/IR/AggregatedOpInterfaceImpl.cpp

```diff
@@ -614,6 +615,61 @@ OnlineAttentionOp::decomposeOperation(OpBuilder &b) {
 // Im2colOp
 //===----------------------------------------------------------------------===//
 
+static std::optional<int64_t>
+chooseDimToVectorize(OpBuilder &b, Location loc, Im2colOp im2colOp,
+                     SmallVector<Range> iterationDomain,
+                     SmallVector<OpFoldResult> inputSizes,
+                     OpFoldResult kOffset) {
+  int64_t innerInputDim = im2colOp.getInputRank() - 1;
+  SmallVector<SmallVector<int64_t>> vectorizationMap =
+      im2colOp.getInputToOutputDimVectorizationMap();
+  SmallVector<int64_t> vectorizableOutputDims = vectorizationMap[innerInputDim];
+  if (vectorizableOutputDims.empty()) {
+    return std::nullopt;
+  }
+  SetVector<int64_t> kDimSet(llvm::from_range, im2colOp.getKOutputDims());
+  // There may be multiple output dims that we can vectorize, so prioritize the
+  // innermost dims first.
+  std::sort(vectorizableOutputDims.begin(), vectorizableOutputDims.end());
+  // Check each dim in order from innermost to outermost, and return the first
+  // one that is vectorizable.
+  while (!vectorizableOutputDims.empty()) {
+    int64_t outputDimToVectorize = vectorizableOutputDims.pop_back_val();
+    // If a K dim is being vectorized, then it is contiguous along either the
+    // input channel dimension, or the filter kernel window. If it is contiguous
+    // along the kernel window, then the actual inner slice size is equal to the
+    // size of the corresponding kernel window dimension. Otherwise, the inner
+    // slice size is just the size of the input tensor's inner dimension.
```

Comment:
I'm a bit confused. If it is contiguous along the kernel window, isn't the kernel window dim also inner dimension? What makes this different from the other case?